### PR TITLE
Fix setting payload customer

### PR DESCRIPTION
**Pull request by ianhalpern**
# Fix Setting Payload Customer ID & Add Comprehensive Test Suite

##  Summary

  This PR addresses customer ID handling issues and introduces a comprehensive
  testing infrastructure for the Payload WooCommerce plugin.

##  Key Changes

  🐛 Bug Fixes

  - Fixed customer ID handling in get_payload_customer_id() function - switched to $user->ID from $user->get_id() which was not a valid function
  - Fixed payment method description in WooCommerce Blocks integration to use method_description instead of non-existent description property
  - Enhanced payment processing with proper transaction descriptions and WooCommerce reference ID mapping
  - Updated `wp_enqueue_script` to load script into footer

##  🧪 Testing Infrastructure

  - Added PHPUnit test suite with 24+ comprehensive tests covering:
    - Payment gateway functionality (payment processing, token creation, subscriptions)
    - WooCommerce Blocks integration
    - Utility functions (customer management, API setup, payment intents)
  - Created mock infrastructure for WordPress, WooCommerce, and Payload APIs to enable isolated testing
  - Added GitHub Actions workflow for automated testing across PHP 8.1, 8.2
  - Integrated code linting with existing development workflow

##  🔧 Development Improvements

  - Enhanced error handling for edge cases in payment processing
  - Added proper namespacing for mock classes to prevent conflicts
  - Improved code quality with comprehensive test coverage
  - Added build and deployment pipeline with artifact generation

##  Test Coverage

  - ✅ Payment processing flows
  - ✅ Subscription handling
  - ✅ Token management and storage
  - ✅ API integration points
  - ✅ Error handling scenarios
  - ✅ WordPress/WooCommerce integration

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    branches: [main, develop]
+
+jobs:
+  php-tests:
+    name: PHP Tests
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version: [8.1, 8.2]
```
Comment (RickRen7575) [inline .github/workflows/tests.yml:16]: Why these two specifically?

File: includes/class-wc-payload-gateway.php [hunk 2]
```diff
@@ -281,6 +283,9 @@ public function create_token( $payment_method ) {
 		$token->set_user_id( get_current_user_id() );
 		$token->save();
 
+		$pm = new Payload\PaymentMethod( array( 'id' => $payment_method['id'] ) );
+		$pm->update( array( 'attrs' => array( '_wp_token_id' => $token->get_id() ) ) );
+
```
Comment (RickRen7575) [inline includes/class-wc-payload-gateway.php:288]: I don't know PHP but would it be better formatting to have this on several lines?

File: includes/class-wc-payload-gateway.php [hunk 3]
```diff
@@ -281,6 +283,9 @@ public function create_token( $payment_method ) {
 		$token->set_user_id( get_current_user_id() );
 		$token->save();
 
+		$pm = new Payload\PaymentMethod( array( 'id' => $payment_method['id'] ) );
```
Comment (RickRen7575) [inline includes/class-wc-payload-gateway.php:286]: What's this `Payload\PaymentMethod` syntax?

**Review by RickRen7575 — APPROVED** [review]
It's a pretty big change, and I don't know PHP very well, but considering it's an urgent fix and you have a plethora of tests that are passing, it looks good to me. Check out the comments I left though.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ianhalpern) [reply .github/workflows/tests.yml:16]: We require at least v8.1 and above and the latest wordpress docker image uses 8.2. We could add 8.3 and 8.4 if we want.

File: includes/class-wc-payload-gateway.php (hunk 3 above)
Comment (ianhalpern) [reply includes/class-wc-payload-gateway.php:286]: This is just initializing the $pm object for updates without retrieving the full object. It's sort of a workaround based on how the php sdk works today.
